LPS-79653 Portlet 3.0: Upgrade to the Portlet 3.0.0 API

ngriffin7a opened the pull request.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

ngriffin7a commented:
ci:test

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest(master)/1296/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-23.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/1296/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test - 0 out of 2 jobs passed in 37 minutes 563 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/428d4905df5a9154eef6377c2ae75f93b3803e89">428d4905df5a9154eef6377c2ae75f93b3803e89</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/ngriffin7a/liferay-portal-ee/tree/LPS-79653-upgrade-portlet3-api">LPS-79653-upgrade-portlet3-api</a><br/>Branch GIT ID: <a href="https://github.com/ngriffin7a/liferay-portal-ee/commit/0deea3e4eb562705d9e2a63d1e7b7ca2431cffb0">0deea3e4eb562705d9e2a63d1e7b7ca2431cffb0</a></p>0 out of 2 jobs PASSED<div><h4>2 Failed Jobs:</h4><ul><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest(master)/1296/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/2368/"><strike><strong>test-portal-acceptance-pullrequest-dist(master)/jdk8</strong></strike></a></li></ul></div><h5>For more details click <a href="https://test-1-23.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/1296/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/2368/">test-portal-acceptance-pullrequest-dist(master)/jdk8</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>6 Tests Failed.</p></div><ol><li><div><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/AXIS_VARIABLE=jboss,label_exp=!master/2368//consoleText">AXIS_VARIABLE=jboss,label_exp=!master #2368</a><pre><code>build-common.compile-test:
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/portal-impl/test-results/integration
     [copy] Copying 101 files to /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [javac] Compiling 398 source files to /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/test/integration/com/liferay/portal/kernel/portlet/bridges/mvc/MVCActionCommandTest.java:143: error: MockLiferayPortletConfig is not abstract and does not override abstract method getPublicRenderParameterDefinitions() in PortletConfig
    [javac] 	private static class MockLiferayPortletConfig
    [javac] 	               ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/test/integration/com/liferay/portal/kernel/portlet/bridges/mvc/MVCActionCommandTest.java:168: error: MockLiferayPortletRequest is not abstract and does not override abstract method getUserAgent() in PortletRequest
    [javac] 	private static class MockLiferayPortletRequest
    [javac] 	               ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 2 errors
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:2053: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:497: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build.xml:297: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/portal-impl/build.xml:513: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-common.xml:1859: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-common.xml:472: The following error occurred while executing this line:</code></pre></div></li><li><div><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/AXIS_VARIABLE=tcserver,label_exp=!master/2368//consoleText">AXIS_VARIABLE=tcserver,label_exp=!master #2368</a><pre><code>build-common.compile-test:
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/portal-impl/test-results/integration
     [copy] Copying 101 files to /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [javac] Compiling 398 source files to /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/test/integration/com/liferay/portal/kernel/portlet/bridges/mvc/MVCActionCommandTest.java:143: error: MockLiferayPortletConfig is not abstract and does not override abstract method getPublicRenderParameterDefinitions() in PortletConfig
    [javac] 	private static class MockLiferayPortletConfig
    [javac] 	               ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/test/integration/com/liferay/portal/kernel/portlet/bridges/mvc/MVCActionCommandTest.java:168: error: MockLiferayPortletRequest is not abstract and does not override abstract method getUserAgent() in PortletRequest
    [javac] 	private static class MockLiferayPortletRequest
    [javac] 	               ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 2 errors
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:2053: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:497: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build.xml:297: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/portal-impl/build.xml:513: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-common.xml:1859: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-common.xml:472: The following error occurred while executing this line:</code></pre></div></li><li><div><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/AXIS_VARIABLE=tomcat,label_exp=!master/2368//consoleText">AXIS_VARIABLE=tomcat,label_exp=!master #2368</a><pre><code>build-common.compile-test:
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/portal-impl/test-results/integration
     [copy] Copying 101 files to /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [javac] Compiling 398 source files to /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/test/integration/com/liferay/portal/kernel/portlet/bridges/mvc/MVCActionCommandTest.java:143: error: MockLiferayPortletConfig is not abstract and does not override abstract method getPublicRenderParameterDefinitions() in PortletConfig
    [javac] 	private static class MockLiferayPortletConfig
    [javac] 	               ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/test/integration/com/liferay/portal/kernel/portlet/bridges/mvc/MVCActionCommandTest.java:168: error: MockLiferayPortletRequest is not abstract and does not override abstract method getUserAgent() in PortletRequest
    [javac] 	private static class MockLiferayPortletRequest
    [javac] 	               ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 2 errors
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:2053: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:497: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build.xml:297: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/portal-impl/build.xml:513: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-common.xml:1859: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-common.xml:472: The following error occurred while executing this line:</code></pre></div></li><li><div><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/AXIS_VARIABLE=weblogic,label_exp=!master/2368//consoleText">AXIS_VARIABLE=weblogic,label_exp=!master #2368</a><pre><code>build-common.compile-test:
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/portal-impl/test-results/integration
     [copy] Copying 101 files to /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [javac] Compiling 398 source files to /opt/dev/projects/github/liferay-portal/portal-impl/test-classes/integration
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/test/integration/com/liferay/portal/kernel/portlet/bridges/mvc/MVCActionCommandTest.java:143: error: MockLiferayPortletConfig is not abstract and does not override abstract method getPublicRenderParameterDefinitions() in PortletConfig
    [javac] 	private static class MockLiferayPortletConfig
    [javac] 	               ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/test/integration/com/liferay/portal/kernel/portlet/bridges/mvc/MVCActionCommandTest.java:168: error: MockLiferayPortletRequest is not abstract and does not override abstract method getUserAgent() in PortletRequest
    [javac] 	private static class MockLiferayPortletRequest
    [javac] 	               ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 2 errors
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:2053: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:497: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build.xml:297: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/portal-impl/build.xml:513: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-common.xml:1859: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-common.xml:472: The following error occurred while executing this line:</code></pre></div></li><li>...</li></ol><strong>Click <a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/2368/testReport">here</a> for more failures.</strong></div></li></ol><h4>For upstream results, click <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream(master)">here</a>.</h4></details></html>

ngriffin7a commented:
Fixed compile failure in a new commit, and added another new commit that adds a cast to PortletURL due to Portlet 3.0 changes to the MimeResponse.createRenderURL() method signature return value type. Submitted new PR, see: https://github.com/shuyangzhou/liferay-portal/pull/6071
